content/en/docs/How To/create_upload.md

```diff
@@ -0,0 +1,85 @@
+---
+title: "Create a custom upload component"
+date: 2020-11-11
+weight: 750
+---
+The last (optional) component of a Baker pipeline is the Upload, whose job is to, precisely,
+upload somewhere what the output component produces.
+
+To create an upload component and make it available to Baker, one must:
+
+* Implement the [Upload](https://pkg.go.dev/github.com/AdRoll/baker#Upload) interface
+* Fill an [`UploadDesc`](https://pkg.go.dev/github.com/AdRoll/baker#UploadDesc) structure and
+register it within Baker via [`Components`](https://pkg.go.dev/github.com/AdRoll/baker#Components).
+
+## The Upload interface
+
+```go
+type Upload interface {
+	Run(upch <-chan string) error
+	Stop()
+	Stats() UploadStats
+}
+```
+
+The [Upload interface](https://pkg.go.dev/github.com/AdRoll/baker#Upload) must be implemented when
+creating a new Upload component.
+
+The `Run` function implements the component logic and receives a channel where the output sends what
+it produces (most probably file paths).
+
+Currently, only the [S3](https://github.com/AdRoll/baker/blob/main/upload/s3.go) component exists
+and it expects to receive in the channel the path to the files produced by an output.
+
+## UploadDesc
+
+```go
+var MyUploadDesc = baker.UploadDesc{
+	Name:   "MyUpload",
+	New:    NewMyUpload,
+	Config: &MyUploadConfig{},
+	Help:   "High-level description of MyUpload",
+}
+```
+
+This object has a `Name`, that is used in the Baker configuration file to identify the upload,
+a costructor-like function (`New`), a config object (where the parsed upload configuration from the
+TOML file is stored) and a help text that must help the users to use the component and its
+configuration parameters.
+
+### Upload constructor-like function
+
+The `New` key in the `UploadDesc` object represents the constructor-like function.
+
+The function receives a [UploadParams](https://pkg.go.dev/github.com/AdRoll/baker#UploadParams)
+object and returns an instance of [Upload](https://pkg.go.dev/github.com/AdRoll/baker#Upload).
+
+The function should verify the configuration params into `UploadParams.DecodedConfig` and initialize
+the component accordingly.
+
+### The upload configuration and help
+
+The upload configuration object (`MyUploadConfig` in the previous example) must export all
+configuration parameters that the user can set in the TOML topology file.
+
+Each key in the struct must include a `help` string tag and a `required` boolean tag, both are
+mandatory.
```

Comment:
`required` is not mandatory. See https://github.com/AdRoll/baker/pull/78/files#r521546397
Same in all PRs actually